src/visitable.cpp

```diff
@@ -103,13 +112,11 @@ static int has_quality_from_vpart( const vehicle& veh, int part, const quality_i
 
             // does the part provide this quality?
             if( iter != tq.end() && iter->second >= level ) {
-                if( ++qty >= limit ) {
-                    break;
```

Comment:
You could be right (due to vectorization) but it could probably be proven via a simple unit test.